src/script/standard.cpp

```diff
@@ -198,6 +205,23 @@ bool ExtractDestination(const CScript& scriptPubKey, CTxDestination& addressRet)
     {
         addressRet = CScriptID(uint160(vSolutions[0]));
         return true;
+    } else if (whichType == TX_WITNESS_V0_KEYHASH) {
+        WitnessV0KeyHash hash;
+        memcpy(hash.begin(), vSolutions[0].data(), 20);
+        addressRet = hash;
+        return true;
+    } else if (whichType == TX_WITNESS_V0_SCRIPTHASH) {
+        WitnessV0ScriptHash hash;
+        memcpy(hash.begin(), vSolutions[0].data(), 32);
+        addressRet = hash;
+        return true;
+    } else if (whichType == TX_WITNESS_UNKNOWN) {
+        WitnessUnknown unk;
+        unk.version = vSolutions[0][0];
+        memcpy(unk.program, vSolutions[1].data(), vSolutions[1].size());
+        unk.length = vSolutions[1].size();
+        addressRet = unk;
+        return true;
```

Comment:
The header comment for `ExtractDestination` is out of date now that P2WPKH and P2WSH are supported, it says

> Currently only works for P2PK, P2PKH, and P2SH scripts.

Reply:
Fixed.